src/GeneticSharp.Domain/Mutations/SelfAdaptiveMutation.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.ComponentModel;
+using System.Linq;
+using GeneticSharp;
+
+namespace GeneticSharp
+{
+    [DisplayName("Self Adaptive Mutation")]
+    public class SelfAdaptiveMutation : MutationBase
+    {
+        public readonly double Tau = 0.1;
+        public readonly double MinMutationRate = 0.05;
+        public readonly double MaxMutationRate = 0.9;
+
+
+        public SelfAdaptiveMutation()
+        {
+        }
+
+        public SelfAdaptiveMutation(double tau = 0.1, double minMutationRate = 0.05, double maxMutationRate = 0.9)
+        {
+            this.Tau = tau;
+            this.MinMutationRate = minMutationRate;
+            this.MaxMutationRate = maxMutationRate;
+        }
+
+        protected override void PerformMutate(IChromosome chromosome, float probability)
+        {
+            if (chromosome is not SelfAdaptiveChromosome adaptiveChromosome)
+            {
+                throw new ArgumentException("The chromosome must be of type SelfAdaptiveChromosome.", nameof(chromosome));
+            }
+
+            var random = RandomizationProvider.Current;
+
+            for (int i = 0; i < adaptiveChromosome.Length; i++)
+            {
+                if (random.GetDouble() < adaptiveChromosome.GetMutationProbability(i))
+                {
+                    // Mutación auto-adaptativa de la tasa de mutación
```

Comment:
Please, only English comments (even though I'm a native Portuguese speaker and can understand a little bit of Spanish, we need to keep all comments in English)